### PR TITLE
Speed up our builds

### PR DESCRIPTION
**Pull request by heytherewill**
### Summary
This is a housekeeping PR. It removes many of the build warning from our builds and enables parallel module builds, which speeds up our compile times a bit.

Looking at the CI, it appears that our builds are now 2 minutes faster

### Technical considerations
There's one warning that I wasn't able to get rid of: `warning: The following options were not recognized by any processor: '[dagger.hilt.android.internal.disableAndroidSuperclassValidation]`

Also, AS 4's build analytizer seems to be not working properly for me. I only get insight on clean operations, but never on the full build 🤷 
If you guys can run it and let me know what you see, I'd appreciate it.

Nonetheless, I tried running ` ./gradlew --profile --rerun-tasks assembleDebug` and the currently slowest tasks are:

:app:mergeDebugJavaResource | 2m21.28s
-- | --
:app:dexBuilderDebug | 1m34.57s 
:app:transformClassesWithFirebasePerformancePluginForDebug | 41.056s 
:app:mergeProjectDexDebug | 19.342s 
:app:mergeDebugNativeLibs | 6.665s 

All other tasks take three seconds or less. If you know any low hanging fruits there as well, let's harvest them. I think we can maybe disable the firebase performance plugin for debug? Maybe?

### Pull Request checklist
- [X] The code does not hinder the app's performance
- [X] The code does not introduce regressions
- [X] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
Please look at it commit by commit

### Review template
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

## Merge permissions
Only myself

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK" />
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="JDK" project-jdk-type="JavaSDK" />
```
Comment (pdrj) [inline .idea/misc.xml:3]: This gives me a `JDK JDK is not defined` error - let's wait for the other macOS guys, but I think we might have to revert it to `1.8` 😛

Comment (heytherewill) [reply .idea/misc.xml:3]: _Sad Windows Noises_

**Comment by daividssilverio** [comment]
@pdrj what is your AS setup? My builds are slow compared to yours 🤔 

My runs after this commit: fc7f59c

Clean, then build:
Screen Shot 2020-07-20 at 11 13 26 AM
Screen Shot 2020-07-20 at 11 15 19 AM

No changes, then build: 0s
No changes, then run app:
Screen Shot 2020-07-20 at 11 19 34 AM
Screen Shot 2020-07-20 at 11 18 56 AM

One line added to the `ContextualMenuFragment`:
Screen Shot 2020-07-20 at 11 21 46 AM
Screen Shot 2020-07-20 at 11 21 35 AM

One action added and reduced in the `ContextualMenuReducer`, then run the app:
Screen Shot 2020-07-20 at 11 26 00 AM
Screen Shot 2020-07-20 at 11 25 44 AM

I'll try master now... should have done that first...

**Comment by daividssilverio** [comment]
On master, ok... the first clean build is faster on master... I might have done something wrong here...

Clean, then build:
Screen Shot 2020-07-20 at 11 38 26 AM
Screen Shot 2020-07-20 at 11 38 16 AM

No changes, build: 6.8s
One line added to the ContextualMenuFragment:
Screen Shot 2020-07-20 at 11 44 02 AM
Screen Shot 2020-07-20 at 11 43 56 AM

One action added and reduced in the ContextualMenuReducer, then run the app:
Screen Shot 2020-07-20 at 11 46 30 AM
Screen Shot 2020-07-20 at 11 46 22 AM

**Comment by daividssilverio** [comment]
@heytherewill @pdrj do you know a better way to run those tests to have more close to reality "benchmark" results?

**Comment by daividssilverio** [comment]
But yeah, just comparing with the other builds on our CI... the changes seem to very welcome.

**Comment by pdrj** [comment]
> @pdrj what is your AS setup? My builds are slow compared to yours 🤔

The default, with some memory tweaks. 
Screenshot 2020-07-21 at 10 20 19

**Comment by pdrj** [comment]
> @heytherewill @pdrj do you know a better way to run those tests to have more close to reality "benchmark" results?

I just did 'Clean project' from the 'Build' menu and then I used the 'Run' green arrow to build and run on device.

**Pull request merged**